docs/trace/README.md

```diff
@@ -1,19 +1,122 @@
 # OpenTelemetry .NET Traces
 
+<!-- markdownlint-disable MD033 -->
+<details>
+<summary>Table of Contents</summary>
+
+* [Best Practices](#best-practices)
+* [Package Version](#package-version)
+* [Tracing API](#tracing-api)
+* [TracerProvider Management](#tracerprovider-management)
+* [Correlation](#correlation)
+
+</details>
+<!-- markdownlint-enable MD033 -->
+
 ## Best Practices
 
-### ActivitySource should be singleton
+The following tutorials have demonstrated the best practices for while using
+traces with OpenTelemetry .NET:
+
+* [Getting Started - ASP.NET Core
+  Application](./getting-started-aspnetcore/README.md)
+* [Getting Started - Console Application](./getting-started-console/README.md)
+
+## Package Version
+
+:heavy_check_mark: You should always use the
+[System.Diagnostics.Activity](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity)
+APIs from the latest stable version of
+[System.Diagnostics.DiagnosticSource](https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/)
+package, regardless of the .NET runtime version being used:
+
+* If you're using the latest stable version of [OpenTelemetry .NET
+  SDK](../../src/OpenTelemetry/README.md), you don't have to worry about the
+  version of `System.Diagnostics.DiagnosticSource` package because it is already
+  taken care of for you via [package
+  dependency](../../Directory.Packages.props).
+* The .NET runtime team is holding a high bar for backward compatibility on
+  `System.Diagnostics.DiagnosticSource` even during major version bumps, so
+  compatibility is not a concern here.
+
+## Tracing API
+
+:stop_sign: You should avoid creating
+[`ActivitySource`](https://learn.microsoft.com/dotnet/api/system.diagnostics.activitysource)
+too frequently. `ActivitySource` is fairly expensive and meant to be reused
+throughout the application. For most applications, it can be modeled as static
+readonly field (e.g. [Program.cs](./getting-started-console/Program.cs)) or
+singleton via dependency injection (e.g.
+[Instrumentation.cs](../../examples/AspNetCore/Instrumentation.cs)).
+
+:heavy_check_mark: You should use dot-separated
+[UpperCamelCase](https://en.wikipedia.org/wiki/Camel_case) as the
```

Comment:
interestingly, Otel spec has no particular guidance for tracer name. The example mentioned in the spec uses full lowercase, though it is separated by dots.
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#get-a-tracer

But for attribute keys, there is convention :  https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/attribute-naming.md

Do we want to open a spec issue, to see if spec has plans to formally recommend conventions for tracer names?

Reply:
> Do we want to open a spec issue, to see if spec has plans to formally recommend conventions for tracer names?

I think the guidance in this PR is a tailored version for .NET users - which means we're good as long as the .NET guidance is not against the spec.

For .NET, I think UpperCamelCase is preferred and it is more consistent with other signals (e.g. Meter, Log Category).

https://github.com/dotnet/runtime/blob/b94a82d63580775b2adb0ccac478719b8cd7049b/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L19

https://github.com/dotnet/runtime/blob/b94a82d63580775b2adb0ccac478719b8cd7049b/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandlerLoggingStrings.cs#L12